### PR TITLE
[4.0] Correcting com_fields created and modified dates

### PR DESCRIPTION
**Pull request by infograf768**
Pull Request for Issue https://github.com/joomla/joomla-cms/issues/24202

### Summary of Changes
Correcting error when creating a new fieldgroup
`Error
Save failed with the following error: Incorrect datetime value: '' for column 'modified' at row 1`

Normalize date fields for Field groups and Fields to use calendar and therefore allow translate format and user_utc, jalali date, etc. ,  on the model of 3.x
Corrected missing code in GroupTable.php (which created the original error)

### Testing Instructions
Create a new Field Group. **No more error after patch**
Create a new Field. Look at the created date field.

### Expected result
Both field groups and fields are created fine.
For fields the created date is now using calendar.
Screen Shot 2019-03-18 at 17 44 01

@bahl24
@SharkyKZ 
@nitesh-solanki

### CODE REVIEW

**Comment by infograf768** [comment]
Note: replaces https://github.com/joomla/joomla-cms/pull/24215

**Comment by SharkyKZ** [comment]
Restore `unset` filter and `readonly` property please. These dates should not be modifiable by user.

**Comment by infograf768** [comment]
Why?
We do not do that in 3.x

**Comment by SharkyKZ** [comment]
Right, that is fine for created date. But not for modified date. Modified date should always be auto generated.

**Comment by infograf768** [comment]
I am confused.
All similar "modified" fields are set with `filter="user_utc"` in core.
They also are set as `readonly` in this patch (as elsewhere) which makes them impossible to modify. They are auto generated OK like that.

**Comment by bahl24** [comment]
@infograf768 Yes, you are right. I was facing the same issue in #24215 . The value being passed is `''` which is not NULL or datetime, therefore default value is not being used.

**Comment by infograf768** [comment]
@bahl24 
Please test this patch and mark your test OK in https://issues.joomla.org/tracker/joomla-cms/24234

**Comment by bahl24** [comment]
I have tested this item :white_check_mark: successfully on 7396aeeea55ee686d26addf7f86a57a12dc051c2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24234">issues.joomla.org/tracker/joomla-cms/24234</a>.</sub>

**Comment by SharkyKZ** [comment]
`readonly` only provides client-side protection, I think. I was able to save a new group with custom modified date.

File: administrator/components/com_fields/Table/GroupTable.php [hunk 1]
```diff
@@ -107,6 +107,11 @@ public function check()
 				$this->created = $date->toSql();
 			}
 
+			if (!(int) $this->modified)
```
Comment (SharkyKZ) [inline administrator/components/com_fields/Table/GroupTable.php:110]: Guess this check should be removed and modified date should be set always.

Comment (SharkyKZ) [reply administrator/components/com_fields/Table/GroupTable.php:110]: And modified date for new items should be `$this->_db->getNullDate()`.

Comment (infograf768) [reply administrator/components/com_fields/Table/GroupTable.php:110]: Here is the difference with 3.x and it was solved this way for FieldTable.php
I just applied the same modification for GroupTable
Please compare the tables in 3.x and 4.0
In 4.0 we have
https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/components/com_fields/Table/FieldTable.php#L151-L154

We could also I guess, use in both cases
`$this->modified = $this->_db->getNullDate();` when the item is new which is the case for this code.

**Comment by infograf768** [comment]
> readonly only provides client-side protection, I think. I was able to save a new group with custom modified date.

How did you do that? 
If you are right, then ALL occurences of modified or modified_time fields should be changed in core.
This is out of this PR scope.

File: administrator/components/com_fields/Table/GroupTable.php (hunk 1 above)
Comment (infograf768) [reply administrator/components/com_fields/Table/GroupTable.php:110]: modified to use `get->getNulldate()`

**Comment by Quy** [comment]
I don't get the error message assuming due to the merge of PR #21901.

**Comment by infograf768** [comment]
@Quy 
Indeed solved for the error but still true for the xmls.
If the xmls are not modified we do not get the `translateformat`
In French we have for example:
Screen Shot 2019-04-01 at 10 40 40

Where the calendar field type exists for the Creation date but not for the Modified date, therefore displaying there the default sql (english format) date.

Will create a new PR just for that.

**Comment by infograf768** [comment]
Closing as this needs a new PR

**Pull request closed without merging**